### PR TITLE
address #44

### PR DESCRIPTION
**Pull request by rsbivand**
@duccioa This is my attempt to sort out the logic, but I'm unsure whether QGIS always uses 3000 when N > 3000, or how the sampling proportion is applied. I'll carry on looking at the QGIS documentation. I'd be grateful for your comments.

### CODE REVIEW

**Review by duccioa — APPROVED** [review]

**Comment by duccioa** [comment]
@rsbivand to me the implementation looks good.
The lines 

```r
if (warnLargeN &&
      (style %in% c("kmeans", "hclust", "bclust", "fisher", "jenks"))) {
      if (nobs > largeN) {
        warning("N is large, and some styles will run very slowly; sampling imposed")
        sampling <- TRUE
# issue 44
        nsamp <- as.integer(ceiling(samp_prop*nobs))
        if (nsamp > largeN) nsamp <- as.integer(largeN)
      }
    }
```
seems to imply that sampling is used in algorithms other than fisher and jenks, but I was not able to find trace of it because nsamp is used only in the fisher and jenks implementations. Am I wrong ?

**Comment by rsbivand** [comment]
@duccioa You are quite right (again), I'll try to introduce `nsamp` to those styles too.

**Comment by rsbivand** [comment]
@duccioa The help page was right, not the code. Adding break finding from a sample in the three other styles is not very obvious, so I just removed those styles from the warning.

File: man/classIntervals.Rd [hunk 1]
```diff
@@ -30,7 +30,7 @@ classIntervals2shingle(x)
   \item{dataPrecision}{default NULL, permits rounding of the interval endpoints (added by Richard Dunlap). The data precision used for printing interval values in the legend returned by \code{findColours}, and in the \code{print} method for classIntervals objects. If intervalClosure is \dQuote{left}, the value returned is \code{ceiling} of the data value multiplied by 10 to the dataPrecision power, divided by 10 to the dataPrecision power. The argument does not round \code{var}, the input variable.}
   \item{warnSmallN}{default TRUE, if FALSE, quietens warning for n >= nobs}
   \item{warnLargeN}{default TRUE, if FALSE large data handling not used}
-  \item{largeN}{default 3000L, the QGIS sampling threshold; over 3000, the observations presented to "fisher" and "jenks" are either a \code{samp_prop=} sample or a sample of 3000, whichever is larger}
+  \item{largeN}{default 3000L, the QGIS sampling threshold; over \code{largeN}, the observations presented to "fisher" and "jenks" are either a \code{samp_prop=} sample or a sample of \code{largeN}, whichever is smaller}
```
Comment (duccioa) [inline man/classIntervals.Rd:33]: @rsbivand I attempted a correction of the documentation:

\item{largeN}{default 3000L, the QGIS sampling threshold; over \code{largeN}, "fisher" and "jenks" are calculated over a sample whose size is either the fraction \code{samp_prop} of the number of observations or \code{largeN}, whichever is smaller}.

Does it make sense ?

Comment (rsbivand) [reply man/classIntervals.Rd:33]: @duccioa Fine, maybe `over \code{largeN}` could be `greater than \code{largeN}`?

Comment (duccioa) [reply man/classIntervals.Rd:33]: Yes, sure!

**Pull request merged**